File: materials/2021-2022/11a/2022-01-05-abstract/circle.cpp
#include "circle.hh"

#include <iostream>

using namespace std;

circle::circle(Point center, int radius) {
  this->center = center;
  this->radius = radius;
}

void circle::print() const {
  cout << "I am circle with center: " << center.to_string()
       << "and radius: " << radius << endl;
}

File: materials/2022-2023/11a/2022-11-04-shapes/Shapes/src/Circle.java
public class Circle implements Shape{
    private int radius;
    Point center;

    public Circle(int radius, Point center) {
        this.radius = radius;
        this.center = center;
    }

    @Override
    public double getPerimeter() {
        return 2 * Math.PI * radius;
    }

    @Override
    public double getArea() {
        return Math.PI * Math.pow(radius, 2);
    }
}
